src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,153 @@
+      ←(continuous_map.tsum_comp_add_apply hf x).tsum_eq,
+    ←tsum_mul_left] },
+  -- Use dominated convergence to rewrite the integral of a sum as a sum of integrals,
+  -- bounding the `n`-th integrand `fₙ` by the sup norm of `f (x + n)` on `[0, 1]`.
+  have := interval_integral.has_sum_integral_of_dominated_convergence
+    (λ (n:ℤ) (x:ℝ), ‖(f.comp_add n).restrict (Icc (0:ℝ) 1)‖) -- (the bound)
+    (λ (n:ℤ), (map_continuous (e * f.comp_add n)).ae_strongly_measurable) -- (measurability)
+    (λ (n:ℤ), ae_of_all _ (λ a ha, _)) -- (nth integrand is bounded by the bound)
+    (ae_of_all _ (λ (x : ℝ) hh, hf ⟨Icc (0:ℝ) 1, is_compact_Icc⟩))  -- (the bound is summable)
+    _ --  (each bound is integrable)
+    (ae_of_all _ (λ x hx, _)), -- sum of f_n is f
+  rw ←this.tsum_eq,
+  rotate, -- first clear all the subsidiary goals from dominated convergence
+  { -- 1/3 : fₙ * e is bounded by the norm of fₙ
+    refine le_trans (le_of_eq _) (continuous_map.norm_coe_le_norm _ _),
+    refine ⟨a, _⟩,
+    { rw uIoc_of_le (zero_le_one' ℝ) at ha,
+      exact mem_of_mem_of_subset ha Ioc_subset_Icc_self },
+    rw [continuous_map.coe_mul, pi.mul_apply, norm_mul, continuous_map.coe_restrict, comp_apply,
+      subtype.coe_mk, continuous_map.comp_add, continuous_map.coe_mk, ne, one_mul] },
+  { -- 2/3 : bound is integrable (because it's constant!)
+    convert interval_integrable_const,
+    ext1 x,
+    simp only,
+    exact real.locally_finite_volume },
+  { -- 3/3: the sum of the fₙ is what it should be
+    convert summable.has_sum _,
+    rw [←summable_norm_iff],
+    simp_rw [continuous_map.coe_mul, pi.mul_apply, norm_mul, ne, one_mul],
+    rw [summable_norm_iff],
+    apply continuous_map.summable_apply,
+    exact continuous_map.summable_of_locally_summable_norm hf },
+  -- Now finish by gathering the sum of integrals into an integral over `ℝ`.
+  dsimp only,
+  convert (_ : integrable (e * f) volume).has_sum_interval_integral_comp_add_int.tsum_eq,
+  { ext1 n,
+    refine interval_integral.integral_congr (λ x hx, _),
+    dsimp only,
+    rw [pi.mul_apply],
+    congr' 1,
+    { dsimp only [e],
+      rw [continuous_map.comp_apply, continuous_map.comp_apply, continuous_map.coe_mk,
+        quotient_add_group.coe_add],
+      suffices : ((n:ℝ) : unit_add_circle) = 0,
+      { rw [this, add_zero], },
+      rw add_circle.coe_eq_zero_iff,
+      use n, simp only [int.smul_one_eq_coe], } },
+  { ext1 x, simp only [fourier_coe_apply, neg_mul, continuous_map.coe_comp, continuous_map.coe_mk,
+      pi.mul_apply, comp_app, int.cast_neg, mul_neg, of_real_one, div_one], },
+  { refine ⟨(map_continuous (e * f)).ae_strongly_measurable, hfi.2.congr' (ae_of_all _ (λ x, _))⟩,
+    rw [pi.mul_apply, norm_mul, ne, one_mul], },
+end
+
+/-- Poisson's summation formula. -/
+theorem poisson_summation {f : C(ℝ, ℂ)}
+  (hf : ∀ (K : compacts ℝ), summable (λ n:ℤ, ‖(f.comp_add n).restrict K‖)) (hfi : integrable f)
+  {g : C(ℝ, ℂ)} (hg : summable (λ n:ℤ, g n))
```

Comment:
You don't need continuity of `g` here, as it follows from the fact that `g` is the Fourier transform of an integrable function.

Reply:
I totally believe you, but proving that here would seem like the wrong place for it -- this is definitely something which belongs in a proper Fourier-transform API.

Reply:
Agreed. What about making a new file `fourier_transform.lean` and proving in there the basic facts you need, with clear documentation that this is only a stub for now? This continuity statement is a trivial consequence of continuity results for integrals, so I would be sad to see it remain as a useless assumption in this statement.

Same thing for `hfi`, by the way: deducing it from `hf` is really straightforward, so IMHO this should definitely be done.

Reply:
> Same thing for `hfi`, by the way: deducing it from `hf` is really straightforward, so IMHO this should definitely be done.

I've just spent over an hour trying to do it and failed, so either I'm very stupid, or it isn't quite as straightforward as you suggest.

Reply:
Sorry, this is clearly not as straightforward as I suggest, then. But this means there are holes in our API, and it is a good occasion to fix them so that it's straightforward the next time. My idea was the following: you need to check ae_measurability (which follows from continuity) and finiteness of the lintegral of the norm. But the lintegral of the norm is the sum of the lintegrals on the intervals [n, n+1) (just like you did for the integral), and so is bounded by the sum of the norms of the translates, which is finite by assumption. Tell me if you don't find a way to implement this, and I'll have a go.

Reply:
That's a good point: I was trying to go via `integrable_of_interval_integral_norm_bounded`, but I got into no end of mathematically trivial but horribly fiddly issues with rearranging sums (tsum vs. sum over finset vs. sum over subtype...). I didn't realise we had `lintegral_Union` to go with `integral_Union`, which suggests an approach more in keeping with the arguments already in the file. I'll see if that works better.